### DISCUSSION TITLE
Query Geometry in Area

### DISCUSSION THREAD

**Discussion by jankrassnigg** [jrouwe/JoltPhysics#667] [Q&A]
I want to query the physics geometry that exists within a certain area (bounding box), because I want to create a navmesh piece-wise at runtime.

I was convinced that I had seen such functionality ages ago. I know that I can query the geometry on a single shape and pass in a bounding box. I thought the same exists as a world query, no?
I guess the alternative is to do an overlap check and then go through all detected shapes individually?

I also somehow always thought that the debug renderer was using this feature to limit debug rendering to some area, but looking into the code, it doesn't seem to do that. Must have imagined that as well, I guess. But I think it would be a useful addition, because debug rendering can become quite slow very quickly, and being able to limit it to a small area around the player may be a useful feature.

**Comment by jrouwe** [comment]
It's a 2 stage thing:

https://github.com/jrouwe/JoltPhysics/blob/153291b7c6e294d727c707fd0734c038cc2fdce0/Samples/SamplesApp.cpp#L1492-L1525

First you query the broadphase for TransformedShapes (takes a read lock so prevents certain other operations from taking place) and then you can get spend a lot of time on a background thread to get the triangles without any additional locking since the TransformedShapes take a reference to the Shape.

**Comment by jankrassnigg** [comment]
That works perfectly, thanks!
